File: assets/bin/Data/Managed/Assembly-CSharp/GetSocialSdk.Ui/InvitesViewBuilder.cs
using GetSocialSdk.Core;
using System;
using UnityEngine;

namespace GetSocialSdk.Ui
{
	public sealed class InvitesViewBuilder : ViewBuilder<InvitesViewBuilder>
	{
		private LinkParams _linkParams;

		private InviteContent _inviteContent;

		private Action<string> _onInviteComplete;

		private Action<string> _onInviteCancel;

		private Action<string, GetSocialError> _onInviteFailure;

		public InvitesViewBuilder SetCustomInviteContent(InviteContent inviteContent)
		{
			_inviteContent = inviteContent;
			return this;
		}

		[Obsolete("Deprecated, use SetLinkParams instead.")]
		public InvitesViewBuilder SetCustomReferralData(CustomReferralData customReferralData)
		{
			_linkParams = new LinkParams(customReferralData);
			return this;
		}

		public InvitesViewBuilder SetLinkParams(LinkParams linkParams)
		{
			_linkParams = linkParams;
			return this;
		}

		public InvitesViewBuilder SetInviteCallbacks(Action<string> onComplete, Action<string> onCancel, Action<string, GetSocialError> onFailure)
		{
			Check.Argument.IsNotNull(onComplete, "onComplete", null);
			Check.Argument.IsNotNull(onCancel, "onCancel", null);
			Check.Argument.IsNotNull(onFailure, "onFailure", null);
			_onInviteComplete = onComplete;
			_onInviteCancel = onCancel;
			_onInviteFailure = onFailure;
			return this;
		}

		internal override bool ShowInternal()
		{
			return ShowBuilder(ToAJO());
		}

		private AndroidJavaObject ToAJO()
		{
			//IL_000b: Unknown result type (might be due to invalid IL or missing references)
			//IL_0010: Expected O, but got Unknown
			AndroidJavaObject val = new AndroidJavaObject("im.getsocial.sdk.ui.invites.InvitesViewBuilder", new object[0]);
			if (_linkParams != null)
			{
				val.CallAJO("setLinkParams", _linkParams.ToAjo());
			}
			if (_inviteContent != null)
			{
				AndroidJavaObject val2 = _inviteContent.ToAjo();
				val.CallAJO("setCustomInviteContent", val2);
			}
			if (_onInviteComplete != null)
			{
				val.CallAJO("setInviteCallback", new InviteUiCallbackProxy(_onInviteComplete, _onInviteCancel, _onInviteFailure));
			}
			return val;
		}
	}
}

File: sources/im/getsocial/sdk/ui/invites/InvitesViewBuilder.java
package im.getsocial.sdk.ui.invites;

import im.getsocial.sdk.internal.p033c.p041b.XdbacJlTDQ;
import im.getsocial.sdk.internal.p047b.upgqDBbsrL;
import im.getsocial.sdk.invites.CustomReferralData;
import im.getsocial.sdk.invites.InviteContent;
import im.getsocial.sdk.invites.LinkParams;
import im.getsocial.sdk.ui.ViewBuilder;
import im.getsocial.sdk.ui.internal.p114i.jjbQypPegg;
import im.getsocial.sdk.ui.invites.p137a.pdwpUtZXDT;

public class InvitesViewBuilder extends ViewBuilder<InvitesViewBuilder> {
    @XdbacJlTDQ
    /* renamed from: d */
    upgqDBbsrL f3228d;
    /* renamed from: e */
    private LinkParams f3229e;
    /* renamed from: f */
    private InviteContent f3230f;
    /* renamed from: g */
    private InviteUiCallback f3231g;

    /* renamed from: a */
    protected final void mo4579a(jjbQypPegg.upgqDBbsrL upgqdbbsrl) {
    }

    /* renamed from: c */
    protected final jjbQypPegg.upgqDBbsrL mo4594c() {
        return new pdwpUtZXDT(new im.getsocial.sdk.ui.invites.p137a.XdbacJlTDQ(), new im.getsocial.sdk.ui.invites.p137a.upgqDBbsrL(this.f3230f, this.f3229e, this.f3231g == null ? null : (InviteUiCallback) this.f3228d.m1060a(InviteUiCallback.class, this.f3231g)));
    }

    public InvitesViewBuilder setCustomInviteContent(InviteContent inviteContent) {
        this.f3230f = inviteContent;
        return this;
    }

    @Deprecated
    public InvitesViewBuilder setCustomReferralData(CustomReferralData customReferralData) {
        LinkParams linkParams = new LinkParams();
        if (customReferralData != null) {
            linkParams.putAll(customReferralData);
        }
        this.f3229e = linkParams;
        return this;
    }

    public InvitesViewBuilder setInviteCallback(InviteUiCallback inviteUiCallback) {
        this.f3231g = inviteUiCallback;
        return this;
    }

    public InvitesViewBuilder setLinkParams(LinkParams linkParams) {
        this.f3229e = linkParams;
        return this;
    }
}
